From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gold856 <117957790+Gold856@users.noreply.github.com>
Date: Thu, 13 Aug 2026 18:15:12 -0400
Subject: [PATCH 10/10] Force integer unit literals to use double storage

Integer unit UDLs with auto or decltype cause the underlying
storage to be int, but we almost always want it to be double.
---
 include/units/core.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/units/core.h b/include/units/core.h
index 8f9c59e407d798421382a0782a4f4fd8974e0eb6..ad714dae8aed1601204abc7d7fab3dabfa8bf4ae 100644
--- a/include/units/core.h
+++ b/include/units/core.h
@@ -263,9 +263,9 @@ namespace units
 		{                                                                                                                                                                                              \
 			return namespaceName::namePlural<double>(static_cast<double>(d));                                                                                                                          \
 		}                                                                                                                                                                                              \
-		constexpr namespaceName::namePlural<int> operator""_##abbreviation(unsigned long long d) noexcept                                                                                              \
+		constexpr namespaceName::namePlural<double> operator""_##abbreviation(unsigned long long d) noexcept                                                                                              \
 		{                                                                                                                                                                                              \
-			return namespaceName::namePlural<int>(static_cast<int>(d));                                                                                                                                \
+			return namespaceName::namePlural<double>(static_cast<double>(d));                                                                                                                                \
 		}                                                                                                                                                                                              \
 	}
 #endif
